docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,487 @@
+    state "<b>Fleet Device (paused)</b><br/><b>owner</b>: the-matching-fleet<br/><b>spec</b>: user writable // management paused" as Fleet_Paused
+    [*] --> Individual_Device
+    Individual_Device --> Fleet_Device : matching fleet labels && owner == nil
+    Fleet_Device --> Individual_Device : no matching fleet labels
+    Fleet_Device --> Fleet_Paused : flightctl.io/fleet-controller=paused
+    Fleet_Paused--> Fleet_Device: ! flightctl.io/fleet-controller=paused
+    Fleet_Paused--> Individual_Device: no matching fleet labels
+```
+
+##### `/device/{name}` with no owner fleet (Individual Device)
+
+Devices with no matching labels to a fleet are manually managed and start with no `metadata.owner`
+reference the fleet controller will not attempt to reconcile or configure the device in any way.
+
+A `metadata.owner` field can be written through the API to associate the device to an external
+owner (like ACM or AAP, etc..), and prevent the fleet controller from managing the device in any way.
+
+If a fleet matches this device, and the device has empty `metadata.owner` the fleet controller
+will claim the device, by setting `metadata.owner` to `Fleet/{fleet-name}`.
+
+The `os.image` and configuration fields in this API support floating tags, and while the fleet manager
+does freeze the tags to hashes, the user is free to setup the device spec with floating references.
+
+###### Expected behaviour regarding external references with floating tags
+
+Changes on external references are monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes to the rendered version which will be exposed
+on the `/device/{name}/rendered` endpoint.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+##### `/device/{name}` while managed by a fleet (Fleet Device)
+
+When managing a device through a fleet, flightctl is responsible
+for maintaining the device spec in sync with the fleet spec; in this case
+*write access is only possible for the modification of labels and other metadata*
+through this API except for the owner fleet.
+
+> [!NOTE]
+> Devices managed by a fleet have their floating tags resolved to stable hashes at this API
+> level. This is done to ensure that the device is always in sync with the fleet spec.
+
+The device will have an owner reference to the managing fleet, i.e.:
+
+```yaml
+  owner: Fleet/optical-inspector-production-fleet
+```
+
+#### `/device/{name}/rendered` (device and user read-only)
+This endpoint is used by the device agent to retrieve the rendered device spec, this endpoint is
+read-only, each device can only access it's own device rendered spec based on the device
+certificate identification, an user can access any device rendered spec for debug purposes.
+
+The agent can pass: `resourceVersion` if this opaque field has not changed the API will return
```

Comment:
"The agent can add the `resourceVersion` query parameter, passing the value from the `metadata.resourceVersion` field of the last device spec it received, in which case the API will..."

I just realised this is inconsistent with K8s, which would return the resource at the exact version requested (or a "410 Gone" error). Should we make this `knownResourceVersion`? Or add a `HEAD /device/{name}/rendered` method?

Reply:
Ok, after talking to avishay this morning I am proposing an updated version. We will expose renderedVersion to the devices. In switch for the current template version.